ubuntu_image/tests/test_main.py

```diff
@@ -398,6 +410,101 @@ def test_does_not_fit(self):
             call('Volume contents do not fit (72B over): '
                  'volumes:<pc>:structure:<mbr> [#0]'))
 
+    def test_hook_fired(self):
+        # For the purpose of testing, we will be using the post-populate-rootfs
+        # hook as we made sure it's still executed as part of of the
+        # DoNothingBuilder.
+        hookdir = self._resources.enter_context(TemporaryDirectory())
+        hookfile = os.path.join(hookdir, 'post-populate-rootfs')
+        # Let's make sure that, with the use of post-populate-rootfs, we can
+        # modify the rootfs contents.
+        with open(hookfile, 'w') as fp:
+            fp.write("""\
```

Comment:
for such snippets of text, you can use "textwrap.dedent" to make it look less bad (it's a cool trick I recently learned):

```
fp.write(textwrap.dedent("""\
                         #!/bin/sh
                         touch $UBUNTU_IMAGE_HOOK_ROOTFS/foo""")
```

Dedent will strip leading whitespace common to all lines.

Reply:
Oh, my comment obviously misses a closing parenthesis. Sorry :/